internal/app/secretless/providers/kubernetes-secrets/provider.go

```diff
@@ -0,0 +1,115 @@
+package kubernetes_secrets
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	plugin_v1 "github.com/cyberark/secretless-broker/pkg/secretless/plugin/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/api/core/v1"
+)
+
+// Provider provides data values from Kubernetes Secrets.
+type Provider struct {
+	Name   string
+	Client *KubeClient
+}
+
+// ProviderFactory constructs a Provider. The API client is configured from
+// in-cluster environment variables and files.
+func ProviderFactory(options plugin_v1.ProviderOptions) plugin_v1.Provider {
+	var client *KubeClient
+	var err error
+	if client, err = NewKubeClient(); err != nil {
+		log.Panicf("ERROR: Could not create Kubernetes Secrets provider: %s", err)
+	}
+
+	provider := Provider{
+		Name:   options.Name,
+		Client: client,
+	}
+
+	return provider
+}
+
+// GetName returns the name of the provider
+func (p Provider) GetName() string {
+	return p.Name
+}
+
+// parseKubernetesSecretID returns the secret id and field name.
+func parseKubernetesSecretID(id string) (string, string) {
+	tokens := strings.SplitN(id, "#", 2)
+
+	return tokens[0], tokens[1]
+}
+
+// GetValue obtains a value by id. Any secret which is stored in Kubernetes Secrets is recognized.
+// The datatype returned by Kubernetes Secrets is map[string][]byte. Therefore this provider needs
+// to know which field to return from the map. The field to be returned is specified by appending '#fieldName' to the id argument.
+func (p Provider) GetValue(id string) ([]byte, error) {
+	id, fieldName := parseKubernetesSecretID(id)
+
+	currentNamespace, err := p.Client.CurrentNamespace()
+	if err != nil {
+		return nil, err
+	}
+
+	secret, err := p.Client.GetSecret(currentNamespace, id);
+	if err != nil {
+		return nil, err
+	}
+
+	value, ok := secret.Data[fieldName];
+	if !ok {
+		return nil, fmt.Errorf("Kubernetes Secrets provider expects the secret '%s' to contain field '%s'", id, fieldName)
```

Comment:
Might need a rewording of the error output - the current one almost makes it sound like it's exclusively a server issue when it could be also to client requesting a field that doesn't exist. I'm thinking something along the lines of  `ERROR: Could not find field '%s'  in Kubernetes secret '%s'` but feel free to make your own.

Reply:
addressed